crates/fuel-core/src/query/subscriptions/test.rs

```diff
@@ -85,6 +85,8 @@ fn squeezed_during_block_production() -> TransactionStatus {
 enum TxStatus {
     /// The transaction has been submitted
     Submitted,
+    /// The transaction has been preconfirmed, but the final status could still be different
+    Preconfirmed,
```

Comment:
Interesting. I guess this is technically true while the block is waiting to be committed. 

I guess it's about how much security the submitter wants to inherit. In our system, there are only extreme situations where the pre-confirmation will be a lie--like the block builder crashes before committing.